subprojects/core/src/integTest/groovy/org/gradle/initialization/BuildInitializationBuildOperationsIntegrationTest.groovy

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.initialization
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationNotificationsFixture
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+
+class BuildInitializationBuildOperationsIntegrationTest extends AbstractIntegrationSpec {
+
+    final buildOperations = new BuildOperationsFixture(executer, temporaryFolder)
+
+    @SuppressWarnings("GroovyUnusedDeclaration")
+    final operationNotificationsFixture = new BuildOperationNotificationsFixture(executer, temporaryFolder)
+
+    def "build operations are fired and build path is exposed"() {
+        buildFile << """
+            task foo {
+                doLast {
+                    println 'foo'
+                }
+            }
+        """
+        when:
+        succeeds('foo')
+
+        then:
+        buildOperations.first(LoadBuildBuildOperationType).details.buildPath == ":"
+        buildOperations.first(LoadBuildBuildOperationType).result.isEmpty()
+
+        buildOperations.first(EvaluateSettingsBuildOperationType).details.buildPath == ":"
+        buildOperations.first(EvaluateSettingsBuildOperationType).result.isEmpty()
+
+        buildOperations.first(LoadProjectsBuildOperationType).details.buildPath == ":"
+        buildOperations.first(LoadProjectsBuildOperationType).result.rootProject.projectDir == settingsFile.parent
+    }
+
```

Comment:
It would be good to have an ignored failing test in here for the composite case, with a link to the issue with why we can't test it.

Reply:
done